FreshAir/RZFUpgradeManager.m

```diff
@@ -93,12 +94,23 @@ - (void)checkForNewUpdate
         BOOL newUnseenVersion = ((lastDisplayed == nil) || ([self.appVersion compare:lastDisplayed options:NSNumericSearch] == NSOrderedAscending));
         BOOL shouldDisplay = ((newVersion && newUnseenVersion) || isForced);
         if (shouldDisplay) {
+            // Check to see if something other than the update prompt is being presented at the moment.
+            if ([self.presentedViewController isKindOfClass:[RZFUpdatePromptViewController class]]) {
+                // If we are already showing an update prompt, bail. There's nothing left to do.
+                return;
+            }
+            else if (self.presentedViewController) {
+                // If any other view controller is being presented (at this moment, only
+                // other option is the release notes), kill it and continue on!
+                [self dismissViewController:self.presentedViewController];
+            }
+
```

Comment:
Is it possible to move this logic into the default implementation of the interaction delegate? I'd prefer to not have any presentation logic in this upgrade manager, so if a consumer wants to use their own implementation of presentation it doesn't have any hidden rules.